test/14-application-manager.spec.coffee

```diff
@@ -191,14 +192,15 @@ describe 'ApplicationManager', ->
 	after ->
 		@applications.images.inspectByName.restore()
 		@applications.docker.getNetworkGateway.restore()
+		@applications.docker.listContainers.restore()
 		Service.extendEnvVars.restore()
 
 	it 'infers a start step when all that changes is a running state', ->
 		Promise.join(
 			@normaliseCurrent(currentState[0])
 			@normaliseTarget(targetState[0], availableImages[0])
```

Comment:
Note that this above calls `normaliseAndExtendAppFromDB`, which now needs to get an `extraState` argument (this is causing some tests to fail)